### DISCUSSION TITLE
編集リクエストの差分表示への要望

### DISCUSSION THREAD

**Discussion by scivola** [increments/qiita-discussions#147] [要望・アイデア・不具合]
編集リクエストを送る際の差分表示についていくつか要望します。

# タイトル，タグも差分が見たい

編集リクエストを送るとき，リクエスト内容が誤っていないか[^q]，かなり気を使います。
そこで，必ず差分表示を見るのですが，タイトルとタグの変更についてはそれができません。できたほうが安心です。

[^q]: この誤りには，例えばキー操作のミスでうっかり文字を加除してしまう，といったことも含まれます。

# 文字単位の差分が見たい

差分が行単位でしか見られません。
長い段落での誤字訂正など，じっくり見ないと分からず，かなりなストレスです。

# 変更箇所を素早く確認したい

長い記事の場合，変更箇所をスクロールで確認するのがやや苦痛です。
とくに，記事が非常に長くて変更箇所がいくつかある場合，あまり速くスクロールすると見落としが怖く，また複数の変更箇所を行ったり戻ったりするとき把握しづらく感じます。

以下のどちらかの機能が実装されれば改善されます。

* 変更箇所とその周辺（前後 5 行とか）のみを表示する
* 「次の変更箇所」「前の変更箇所」にジャンプする機能

**Comment by scivola** [comment]
タイトルの差分が見たい，に関して補足です。

さきほどこんなことがありました。
マイページの「編集リクエスト管理」を見たところ，「下書き」の先頭に，昨日作った編集リクエストがありました。
「あれ？　送ってなかった？　これ何だっけ？」と開いたのですが，差分を見ても何もありません。
タイトルの誤字訂正だったことを思い出しましたが，タイトルも差分が出れば直ちに分かったはずのことでした。

**Comment by kyntk** [comment]
@scivola さん

編集リクエストの差分ついて色々とご意見いただきありがとうございます！

> タイトル，タグも差分が見たい

こちらについて、私としてもその通りだと思います。
特に補足いただいた編集リクエストの確認においては、特に分かりづらいと思いましたので改善します。
ただ、送信前の編集時にタイトルとタグの差分を見せることについては、見せ方が難しそうで、こちらは見せ方等を検討させていただきます。:bow: 

> 文字単位の差分が見たい

こちらについては、編集リクエストの確認ページでは表示されるものの、送信前には表示がされなくなっておりました。
こちらはすぐ修正いたします。:bow: 

> 変更箇所を素早く確認したい

こちらについては「変更箇所とその周辺（前後 5 行とか）のみを表示する」などぜひやりたいと思うので、検討いたします!

**Comment by ohakutsu** [comment]
@scivola 

おまたせしました！

- 文字単位の差分が見たい
- 変更箇所を素早く確認したい

の修正が完了しました！
リリースノートはこちらです 💁‍♂️ 
https://qiita.com/release-notes#%E8%A8%98%E4%BA%8B%E8%B3%AA%E5%95%8F%E3%82%A8%E3%83%87%E3%82%A3%E3%82%BF%E4%B8%8A%E3%81%A7%E5%A4%89%E6%9B%B4%E5%89%8D%E5%BE%8C%E3%81%AE%E5%B7%AE%E5%88%86%E3%81%8C%E8%A6%8B%E3%82%84%E3%81%99%E3%81%8F%E3%81%AA%E3%82%8A%E3%81%BE%E3%81%99
